lib/linter/linter.js

```diff
@@ -1282,9 +1282,20 @@ class Linter {
      * @param {"flat"|"eslintrc"} [config.configType="flat"] the type of config used.
      */
     constructor({ cwd, configType = "flat", flags = [] } = {}) {
+
+        flags.forEach(flag => {
+            if (inactiveFlags.has(flag)) {
+                throw new Error(`The flag '${flag}' is inactive: ${inactiveFlags.get(flag)}`);
+            }
+
+            if (!activeFlags.has(flag)) {
+                throw new Error(`Unknown flag '${flag}'.`);
+            }
+        });
```

Comment:
We also already have these checks in cli.js, where unknown flags result in exit code 2 but inactive flags only log a warning:

https://github.com/eslint/eslint/blob/53b1ff047948e36682fade502c949f4e371e53cd/lib/cli.js#L492-L508

Now, for inactive flags passed on the command line, ESLint would log both a warning (from cli.js) and an error (from linter):

```
$ node bin/eslint Makefile.js --flag test_only_old
InactiveFlag: The 'test_only_old' flag is no longer active: Used only for testing.

Oops! Something went wrong! :(

ESLint: 9.7.0

Error: The flag 'test_only_old' is inactive: Used only for testing.
    at C:\projects\eslint\lib\linter\linter.js:1288:23
    at Array.forEach (<anonymous>)
    at new Linter (C:\projects\eslint\lib\linter\linter.js:1286:15)
    at new ESLint (C:\projects\eslint\lib\eslint\eslint.js:596:24)
    at Object.execute (C:\projects\eslint\lib\cli.js:510:24)
    at async main (C:\projects\eslint\bin\eslint.js:153:22)
```

Perhaps we should sync the behavior to avoid duplicate logs, or maybe just remove the checks in cli.js (if we don't expect to have any cli-level-only flags).

Reply:
Also, these two tests are failing (at least for me locally, CI is broken at the moment) because now they result in throwing an error:

https://github.com/eslint/eslint/blob/53b1ff047948e36682fade502c949f4e371e53cd/tests/lib/cli.js#L1911-L1923

https://github.com/eslint/eslint/blob/53b1ff047948e36682fade502c949f4e371e53cd/tests/lib/eslint/eslint.js#L330-L334

Reply:
Thanks, I'll update these.